src/main/java/network/brightspots/rcv/Tabulator.java

```diff
@@ -918,29 +944,52 @@ private void recordSelectionForCastVoteRecord(
 
     cvr.setCurrentRecipientOfVote(selectedCandidate);
     if (selectedCandidate == null) {
-      cvr.exhaust();
+      cvr.exhaustBy(statusForRound);
+    }
+
+    if (statusForRound != StatusForRound.ACTIVE) {
+      currentRound.addInactiveBallot(statusForRound, cvr.getFractionalTransferValue());
+    }
+
+    String outcomeDescription;
+    switch (statusForRound) {
+      case ACTIVE ->
+          outcomeDescription = selectedCandidate;
+      case INACTIVE_BY_UNDERVOTE ->
+          outcomeDescription = "underote" + additionalLogText;
+      case INACTIVE_BY_OVERVOTE ->
+          outcomeDescription = "overvote" + additionalLogText;
+      case INACTIVE_BY_SKIPPED_RANKING ->
+          outcomeDescription = "exhausted by skipped ranking" + additionalLogText;
+      case INACTIVE_BY_REPEATED_RANKING ->
+          outcomeDescription = "duplicate candidate" + additionalLogText;
+      case INACTIVE_BY_EXHAUSTED_CHOICES ->
+          outcomeDescription = "no continuing candidate" + additionalLogText;
+      // When we encounter a continuing candidate, we record it with this string in the output logs
+      default ->
+        throw new RuntimeException("Unexpected ballot status: " + statusForRound);
```

Comment:
this is basically an assertion, right? I think you guys changed our assertions to throws in a recent PR, but I think in cases where we're trying to catch a bug in our code vs. handling bad input data, we should at least make that clear with an inline comment (or in the exception message)